### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by Metsanpeitto**
In this branch I had implemented the following features: 
- Add unit tests to your classes:
  - `Book`
  - `Person`
  - `Student`
  - `Teacher`
  - `Classroom`
  - `Corrector`
  - `Rental`

### CODE REVIEW

File: RSpec/spec/classroom_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require './classes/classroom'
+
+describe Classroom do
+  context 'Given arguments to initialize the class' do
+    it 'create a new class giving it a label' do
+      classroom = Classroom.new('4F')
+      expect(classroom.label).to(eq('4F'))
+    end
+  end
+end
```
Comment (Emmyn5600) [inline RSpec/spec/classroom_spec.rb]: - [ ] In the classroom spec, let us add the following tests

    - Adds new student in the classroom
    - Check classroom of newly added student.

File: RSpec/spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      person = Person.new(age: 33)
+      expect(person.name).to(eq('Unknown'))
+    end
+    it 'sets the new Person\'s parent permission to \'true\' if not provided' do
+      person = Person.new(age: 33)
+      expect(person.parent_permission).to(eq(true))
+    end
+    it 'rais an error if  age is not provided' do
+      expect { Person.new }.to(raise_error)
+    end
```
Comment (Emmyn5600) [inline RSpec/spec/person_spec.rb]: - [ ] Optional, In this test, we are facing this warning, try to see if you can fix it.
ruby-24

File: RSpec/spec/rental_spec.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  context 'Given arguments to initialize the class' do
+    it 'create a rental with date, book title and name of the person who rents it ' do
+      rental = Rental.new(Time.now.to_i)
+      book = Book.new({ title: 'The title', author: 'AUTHOR' })
+      person = Person.new(age: 24, parent_permission: true, name: 'Janni')
+      rental.book = book
+      rental.person = person
+      expect(rental.book).to(eq('The title'))
+    end
+  end
```
Comment (Emmyn5600) [inline RSpec/spec/rental_spec.rb]: - [ ] In the rental, we are missing this,  it `should set the correct date`

File: RSpec/spec/student_spec.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+require './classes/student'
+require './classes/person'
+
+describe Student do
+  context 'Given arguments to initialize the class' do
+    it 'create a student' do
+      student = Student.new(name: 'abdelgahnykjkjkjkjkjkjkjkjkjkjkjkjkjkjkjkj', age: 50, classroom: '4f')
+      expect(student.play_hookey).to(eq("¯\(ツ)/¯"))
+    end
+  end
```
Comment (Emmyn5600) [inline RSpec/spec/student_spec.rb]: - [ ] In the student_spec, let us implement the following:

    - returns classroom of student
    - returns id of student

File: RSpec/spec/teacher_spec.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    # it 'serializes object' do
+    #   expect(subject.person).to eq({
+    #                                  specialization: 'Foo bar'
+    #                                })
+    # end
+    # it 'create a teacher with specialization as argument ' do
+    #  teacher = Teacher.new('Science')
+    #  expect(teacher.play_hookey).to(eq("¯\(ツ)/¯"))
+    # end
+  end
```
Comment (Emmyn5600) [inline RSpec/spec/teacher_spec.rb]: - [ ] Let us remove the commented lines

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED :recycle:

### Hi, @Metsanpeitto and @elasfarc 

Great work on implementing the project requirements, however there are some issues to be solved before you get approved but you are almost there:muscle:
  
May I highlight the following:
- There are no linter errors:heavy_check_mark:
- The git-flow has been followed :heavy_check_mark:
- The repo looks professional:heavy_check_mark:
- Tests are running successfully:heavy_check_mark:

### Changes requested.

- _Add another contributor to the readme file_
- _In the Readme file, kindly guide us how to run the tests._
- _Check the comments under the readme file_


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: RSpec/spec/teacher_spec.rb (hunk 5 above)
Comment (Metsanpeitto) [reply RSpec/spec/teacher_spec.rb]: Ok. I fixed it :heavy_check_mark:

File: RSpec/spec/student_spec.rb (hunk 4 above)
Comment (Metsanpeitto) [reply RSpec/spec/student_spec.rb]: Ok. It is fixed now :heavy_check_mark:

File: RSpec/spec/rental_spec.rb (hunk 3 above)
Comment (Metsanpeitto) [reply RSpec/spec/rental_spec.rb]: Perfect. Fixed ! :heavy_check_mark:

File: RSpec/spec/person_spec.rb (hunk 2 above)
Comment (Metsanpeitto) [reply RSpec/spec/person_spec.rb]: Problem solved ! :heavy_check_mark:

File: RSpec/spec/classroom_spec.rb (hunk 1 above)
Comment (Metsanpeitto) [reply RSpec/spec/classroom_spec.rb]: Done ! :heavy_check_mark:

**Review by ElijahTobs — APPROVED** [review]
# STATUS: APPROVED 🎉 🥇 
## Hi Team 👋  

Good job making the changes required by the previous Code Reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


_If anything doesn't seem clear to you, feel free to leave your question or comment in the PR thread below_

#### "Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise."

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

_Reviewed by_: __**Elijah Ayandokun**__

**Pull request merged**